src/Admin/CitizenProjectAdmin.php

```diff
@@ -277,9 +292,84 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
+                'label' => 'Prénom du créateur',
+                'show_filter' => true,
+                'field_type' => TextType::class,
+                'callback' => function (ProxyQuery $qb, string $alias, string $field, array $value) {
+                    if (!$value['value']) {
+                        return;
+                    }
+
+                    /* @var ProxyQuery|QueryBuilder $qb */
+                    $qb
+                        ->leftJoin(Adherent::class, 'creator', Join::WITH, sprintf('creator.uuid=%s.createdBy', $alias))
+                        ->andWhere(
+                            'LOWER(creator.firstName) LIKE :firstName '.
+                            "OR $alias.createdBy=:uuid"
+                        )
+                        ->setParameter('firstName', sprintf('%%%s%%', mb_strtolower($value['value'])))
+                        ->setParameter('uuid', $value['value']);
+
+                    return true;
+                },
+            ])
+            ->add('creatorLastName', CallbackFilter::class, [
+                'label' => 'Nom du créateur',
+                'show_filter' => true,
+                'field_type' => TextType::class,
+                'callback' => function (ProxyQuery $qb, string $alias, string $field, array $value) {
+                    if (!$value['value']) {
+                        return;
+                    }
+
+                    /** @var ProxyQuery|QueryBuilder $qb */
+                    if (!in_array('creator', $qb->getAllAliases(), true)) {
+                        $qb->leftJoin(Adherent::class, 'creator', Join::WITH, sprintf('creator.uuid=%s.createdBy', $alias));
+                    }
+
+                    $qb->andWhere(
+                            'LOWER(creator.lastName) LIKE :lastName '.
+                            "OR $alias.createdBy=:uuid"
+                        )
+                        ->setParameter('lastName', sprintf('%%%s%%', mb_strtolower($value['value'])))
+                        ->setParameter('uuid', $value['value']);
+
+                    return true;
+                },
+            ])
+            ->add('creatorEmailAddress', CallbackFilter::class, [
+                'label' => 'Email du créateur',
+                'show_filter' => true,
+                'field_type' => EmailType::class,
+                'callback' => function (ProxyQuery $qb, string $alias, string $field, array $value) {
+                    if (!$value['value']) {
+                        return;
+                    }
+
+                    /** @var ProxyQuery|QueryBuilder $qb */
+                    if (!in_array('creator', $qb->getAllAliases(), true)) {
+                        $qb->leftJoin(Adherent::class, 'creator', Join::WITH, sprintf('creator.uuid=%s.createdBy', $alias));
+                    }
+
+                    $qb->andWhere(
+                            'LOWER(creator.emailAddress)=:emailAddress '.
+                            "OR $alias.createdBy=:uuid"
```

Comment:
it's really necessary to search an email (`EmailType` field) in uuid column?

Reply:
It's the way it is done and similar other field so I stick to it

Reply:
why? 😕 
the input[type=email] field allows you to insert only a valid email address and it makes no sense to chech UUID column with an email.

Reply:
I misunderstood what was done in the other fields.. all fixed now. Thank you very much :-)